[Bug] Failed to build FE on CentOS 6

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

Version: master

### What's Wrong?

```shell
[INFO]     Processing (grpc-java): olap_file.proto
[INFO]     Plugin path: /tmp/protoc-gen-grpc-java7667690249817629192.exe
protoc-jar: executing: [/Programs/doris/thirdparty/installed/bin/protoc, -I/Programs/doris/fe/fe-core/../../gensrc/proto, --grpc-java_out=/Programs/doris/fe/fe-core/target/generated-sources, --plugin=protoc-gen-grpc-java=/tmp/protoc-gen-grpc-java7667690249817629192.exe, /Programs/doris/fe/fe-core/../../gensrc/proto/olap_file.proto]
/tmp/protoc-gen-grpc-java7667690249817629192.exe: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /tmp/protoc-gen-grpc-java7667690249817629192.exe)
--grpc-java_out: protoc-gen-grpc-java: Plugin failed with status code 1.
[ERROR] /Programs/doris/fe/fe-core/../../gensrc/proto/olap_file.proto [0:0]: /tmp/protoc-gen-grpc-java7667690249817629192.exe: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by /tmp/protoc-gen-grpc-java7667690249817629192.exe)
[ERROR] /Programs/doris/fe/fe-core/../../gensrc/proto/olap_file.proto [0:0]: --grpc-java_out: protoc-gen-grpc-java: Plugin failed with status code 1.
[ERROR] /Programs/doris/fe/fe-core/../../gensrc/proto/olap_file.proto [0:0]:
```

### What You Expected?

Build FE successfully on CentOS 6.

### How to Reproduce?

```shell
./build.sh --fe --clean
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/fe/fe-core/pom.xml
+++ b/fe/fe-core/pom.xml
@@ -70,7 +70,7 @@ under the License.
             </activation>
             <properties>
                 <protoc.artifact>com.google.protobuf:protoc:${protobuf.version}:exe:osx-x86_64</protoc.artifact>
-                <grpc.java.artifact>io.grpc:protoc-gen-grpc-java:${grpc.version}:exe:osx-x86_64</grpc.java.artifact>
+                <grpc.java.artifact>io.grpc:protoc-gen-grpc-java:${grpc-java.version}:exe:osx-x86_64</grpc.java.artifact>
             </properties>
         </profile>
         <profile>

--- a/fe/pom.xml
+++ b/fe/pom.xml
@@ -238,6 +238,8 @@ under the License.
         <!--The dependence of transitive dependence cannot be ruled out, only Saving the nation through twisted ways.-->
         <netty-3-test.version>3.10.6.Final</netty-3-test.version>
         <objenesis.version>2.1</objenesis.version>
+        <!-- NOTE: Using grpc-java whose version is newer than 1.34.0 will break the build on CentOS 6 due to the obsolete GLIBC -->
+        <grpc-java.version>1.34.0</grpc-java.version>
         <grpc.version>1.58.0</grpc.version>
         <check.freamework.version>3.38.0</check.freamework.version>
         <protobuf.version>3.24.3</protobuf.version>
@@ -245,7 +247,7 @@ under the License.
         <!-- see https://repo.maven.apache.org/maven2/com/google/protobuf/protoc/ to get correct version -->
         <protoc.artifact.version>3.24.3</protoc.artifact.version>
         <protoc.artifact>com.google.protobuf:protoc:${protoc.artifact.version}</protoc.artifact>
-        <grpc.java.artifact>io.grpc:protoc-gen-grpc-java:${grpc.version}</grpc.java.artifact>
+        <grpc.java.artifact>io.grpc:protoc-gen-grpc-java:${grpc-java.version}</grpc.java.artifact>
         <protoparser.version>3.1.5</protoparser.version>
         <snappy-java.version>1.1.10.1</snappy-java.version>
         <automaton.version>1.11-8</automaton.version>

